test/gateway/ost_prime/wrap.js

```diff
@@ -0,0 +1,209 @@
+const OSTPrime = artifacts.require("TestOSTPrime")
+  , BN = require('bn.js');
+
+const Utils = require('../../../test/test_lib/utils');
+const EventDecoder = require('../../test_lib/event_decoder.js');
+
+contract('OSTPrime.sol', function (accounts) {
+
+  describe('Unwrap', async () => {
+
+    const DECIMAL = new BN(10);
+    const POW = new BN(18);
+    const DECIMAL_FACTOR = DECIMAL.pow(POW);
+    const TOKENS_MAX = new BN(800000000).mul(DECIMAL_FACTOR);
+
+    let brandedTokenAddress, ostPrime, callerAddress, amount;
+
+    async function initialize(){
+      await ostPrime.initialize(
+        {from: accounts[5], value:TOKENS_MAX}
+      );
+    };
+
+    beforeEach(async function () {
+
+      brandedTokenAddress = accounts[2];
+      ostPrime = await OSTPrime.new(brandedTokenAddress);
+
+      callerAddress = accounts[3];
+      amount = new BN(500);
```

Comment:
It would be relevant for this test what the balance of `accounts[3]` is. Unfortunately, nothing can be assumed as other tests might have changed the balance before this test runs. It is not well decoupled that way.

Reply:
True!
Accounts have 10000000000000000000000000000000 balance.
Which is still enough considering 800M ether as max gas to run the tests (Considering other tests are just consuming gas and not moving the balance in large amount).

Also the tests are considering the initial balances while asserting the final balances.